aip/general/0193.md

```diff
@@ -75,224 +84,354 @@ following:
 }
 ```
 
-### Details
 
-Google defines a set of [standard detail payloads][details] for error details,
-which cover most common needs for API errors. Services **should** use these
-standard details payloads when feasible.
+#### google.rpc.Status
 
-Structured details with machine-readable identifiers **must** be used if users
-will need to write code against a specific aspect of the error. Error message
-strings **may** change over time; however, if an error message does not have a
-machine-readable identifier _in addition to_ the `code` field, changing the
-error message **must** be considered a backwards-incompatible change.
+Services must return a [`google.rpc.Status`][Status] message when an API error occurs. For each error the service **must** populate the `message` string in
+[`google.rpc.Status`][Status] as follows:
 
-#### ErrorInfo
++ present a developer-facing, human-readable "debug message"
++ present the message in the service's native language to explain both the error
+  and offer an actionable resolution to it ([citation][Error model])
 
-The [`ErrorInfo`][ErrorInfo] message is the required way to send a
-machine-readable identifier. All error responses **must** include an
-`ErrorInfo` payload in the `details` field. Variable information
-**should** be included in the `metadata` field on `ErrorInfo` and
-**must** be included if it appears within an error message.
+**Note:** Sometimes a service will elect to always present `Status.message` in English rather than the application's native language so that messages are easily searchable in common knowledge bases, such as StackOverflow™.
 
-#### Uniqueness
+Important points:
 
-Each type of detail payload **must** be included at most once. For
-example, there **must not** be more than one [`BadRequest`][BadRequest]
-message in the `details` field, but there **may** be a `BadRequest` and
-a [`PreconditionFailure`][PreconditionFailure].
++ This `message` string is considered a problem description. It is intended for
+  human developers to understand the problem and is more detailed than
```

Comment:
I would just say "developers" instead of "human developers"

Reply:
Ack.